board cpu and core: header file include guards repaired

rajma996 opened the pull request:
pull request created as per the issue https://github.com/RIOT-OS/RIOT/issues/2623
changes made in board directory and some files in cpu directory

gebart commented on boards/chronos/include/board.h:
```diff
@@ -18,8 +18,8 @@
-#ifndef CHRONOS_BOARD_H_
-#define CHRONOS_BOARD_H_
+#ifndef CHRONOS_BOARD_H
+#define CHRONOS_BOARD__
```
something went wrong here.

rajma996 replied:
Thanks for pointing, it's resolved now.

gebart commented on cpu/sam3/include/sam3x8e.h:
```diff
@@ -27,8 +27,8 @@
-#ifndef _SAM3X8E_
-#define _SAM3X8E_
+#ifndef SAM3X8E
+#define SAM3X8E
```
add a trailing `_H` to match the other include guards

rajma996 replied:
Thanks for pointing, it's resolved now.

gebart commented:
I am not sure of the implications of the changes to the CMSIS headers (`cpu/cortexm_common/include/core_*.h`), and `__CMSIS_GENERIC` is not a traditional include guard and should be left as-is.
Changing the CMSIS headers increases maintenance when upgrading CMSIS releases as well so I propose to just leave them as they are right now.

The SAMxxx headers are missing a trailing `_H` to match the pattern of all other include guards in the tree.

Kijewski commented:
rm  core/include/cscope.out

BytesGalore commented:
I'm with @gebart, since the CMSIS headers define an open standard they are subject of changes and probably will change in the future.
So we should keep them untouched for seamless upgrade and maintenance.

BytesGalore commented on sys/include/pipe.h:
```diff
@@ -32,8 +32,8 @@
-#define __PIPE__H
+#ifndef PIPE__H
+#define PIPE__H
 
```
could you change also the `#endif` in [L122](https://github.com/rajma996/RIOT/blob/master/sys/include/pipe.h#L122) to `#endif /* PIPE__H */`?

BytesGalore commented:
@rajma996 beside the discussion on the CMSIS headers and my request for adding a comment your changes looks good :+1: 
Since you started to adjust the headers in `sys/` I propose to make a separate PR for them and go for merging this PR soon.
So I would say please address our requests and squash your commits.

rajma996 commented:
@BytesGalore Thanks for the feedback. CMSIS headers are restored. As I will be making a separate PR for /sys headers,  changes in the sys/include/pipe.h will be done there :)

BytesGalore commented:
@rajma996 I would say, please squash the 15 commits of this RP into a single one :)
And then lets see what travis says.

BytesGalore commented:
@rajma996 please rebase and squash the commits :)

rajma996 commented:
@BytesGalore  sorry for being late, I tried some wierd method to sync with the main repo... I've reverted back to the last header updating commit. should I squash from the commit I first started  editing headers.

BytesGalore commented:
@rajma996 don't know if you strictly separated your changes in the individual commits, but it would be nice if you could squash to 3 commits like:
* `boards: repaired header guards`
* `cpu: repaired header guards`
* `core: repaired header guards`

If this is not possible, no problem :) then just squash all your changes to one commit and rename it accordingly, e.g. `boards/cpu/core: repaired header guards`.
You also need to rebase this PR to the current master,  if you have troubles with it just ask :)

rajma996 commented:
@BytesGalore thanks for the reply. I reverted all the commits I used for syncing with the main repo previously and squashed everything from the beginning of this forked repo. Let me try rebasing with the current master. I will keep you updated. Again thanks :D

rajma996 commented:
@BytesGalore though changes in the commit looks fine, but Travis checks have failed....:(

BytesGalore commented on cpu/stm32f1/include/stm32f103xe.h:
```diff
@@ -6842,7 +6842,11 @@ typedef struct
   }
 #endif /* __cplusplus */
 
+<<<<<<< HEAD:cpu/stm32f1/include/stm32f103xe.h
```
I guess this one slipped through :) 
(That's what travis complains)

BytesGalore commented:
oh and please `reword` the commit message to something like:
`board cpu core:  header file include guards repaired`

BytesGalore commented:
cool :+1: then lets kick travis

BytesGalore commented on cpu/atmega_common/avr-libc-extra/errno.h:
```diff
@@ -29,8 +29,8 @@
-#define __ERRNO_H_ 1
+#ifndef ERRNO_H
+#define ERRNO_H 1
 
```
I guess this guy needs to remain with underscores

BytesGalore commented on cpu/cortexm_common/include/core_cm0.h:
```diff
@@ -39,8 +39,8 @@
-#ifndef __CORE_CM0_H_GENERIC
-#define __CORE_CM0_H_GENERIC
+#ifndef CORE_CM0_H_GENERIC
+#define CORE_CM0_H_GENERIC
```
oh, and all CMSIS headers `cpu/cortexm_common/include/core_*.h` must remain untouched.

rajma996 replied:
@BytesGalore how do we get to know these errors from travis build....?

BytesGalore replied:
You mean how to see what went wrong with the travis build right?

You can click on [Details](https://travis-ci.org/RIOT-OS/RIOT/builds/109902000) in the CI "info-box" below to get an overview of your (this) build.
There you see a list of build jobs for this PR.
If you click one which is in progress or is already finished/failed you will be presented with the output of the specific build.
Usually the important information can be found at the bottom, giving a hint what actually went wrong.

rajma996 commented:
@BytesGalore build is successful this time.....:)

BytesGalore commented:
nice one :+1:, but could you revert the changes in the CMSIS headers:
``` 
cpu/cortexm_common/include/core_cm0.h
cpu/cortexm_common/include/core_cm0plus.h
cpu/cortexm_common/include/core_cm3.h
cpu/cortexm_common/include/core_cm4.h
cpu/cortexm_common/include/core_cm7.h
```
You can then directly squash the changes.
Were pretty close merging it :)
When this is done, and travis is happy I give my ACK.

rajma996 commented:
@BytesGalore build wasn't successful this time, but i am not able to understand the error.... looks like it couldn't connect for updates.

BytesGalore commented:
@rajma996 seems unrelated I kicked travis to repeat the failed build

rajma996 commented:
@BytesGalore  build is successful. :D

BytesGalore commented:
nice one, ACK from my side :+1: 
Since this PR touches so much files, also in `core`, I would like to have another ACK before we merge it.
@gebart, @PeterKietzmann would you take a look?

rajma996 commented:
@BytesGalore Based on previous requests on PR [4616](https://github.com/RIOT-OS/RIOT/pull/4616) I should've made separate commit for each directory instead of squashing all into one. Let me work on that now...

authmillenon commented:
Please rebase.

authmillenon commented:
Still needs rebase.

rajma996 commented:
@BytesGalore  I am a bit stuck here, I tried rebasing but wasn't successful :( Can you help...

BytesGalore commented:
@rajma996 ok I see your changes are now made on your `master` branch, which is not what you want.
It seems you additionally merged commits/changes from RIOT `master` to your commits.
But don't worry we can manage it :)  

The goal is to keep the changes on the 109 headers but throw all other changes to get in sync with the original RIOT `master` again.
So we first clone your RIOT, checkout the `master` branch and switch to the RIOT directory to start.
(*Just to prevent uncontrolled heat/sweating, backup the folder before starting.*)

1. We remember the filenames of the 109 headers you edited:  
`git diff --name-only b986ac33e81f89acf07a6ca814b7cd7cdffb109a >> ../keep_em_all.txt`  
This will create a file `keep_em_all.txt` in the above directory containing the relative paths + filenames of the 109 headers.
2. we go back **2** commits, since your commit seems to be merged on top of the former one:  
`git reset HEAD~2`  
If you type `git show` it should list the changes for your 109 edited header files plus a large bunch of other changed files + untracked ones.  
If you type `git log` you should see commit `b986ac33e81f89acf07a6ca814b7cd7cdffb109a` on top, which is the first one in sync with RIOT.
3. we open a new branch to move our changes and clean-up `main`:  
`git checkout -b repaired_header_guards`  
4. now we add only the 109 headers we kept in `keep_em_all.txt` to commit them:  
`for word in $(cat ../keep_em_all.txt); do git add $word; done`  
If you type `git status` you should see (when you scroll some pages up) the 109 header files added for being committed.
5. Then lets commit them:  
`git commit -m "core/board/cpu: repaired header guards"`  
If you type `git status` again, only the other changed/untracked files will be listed.
6. Now we want to get rid of them, so we `checkout` their original versions:   
`for word in $(git diff --name-only b986ac33e81f89acf07a6ca814b7cd7cdffb109a); do git checkout $word; done`  
This time we didn't needed to remember their filenames, so we processed them directly.
If you type `git status` you should see only as untracked marked files.
7. Now we switch to `master` to clean it up completely:  
`git checkout master`  
and then `stash` the untracked marked files:  
`git stash`  
stashes the files and finally allows us to sync your `master` with the original RIOT `master`
(basically you can delete the stash afterwards since the changes are introduced back after rebase)
8. Now we rebase your `master`:  
`git pull --rebase https://github.com/RIOT-OS/RIOT.git`
9. After that you can rebase your branch `repaired_header_guards`:  
`git checkout repaired_header_guards`
`git rebase master`  
fixing appearing conflicts... ... ...
10. and finally pushing it to your repo:   
`git push -u origin repaired_header_guards`  
and  
`git push -f` for your `master` branch.  
**Note:** the `-f` forces to overwrite your remote repo stuff with the local changes, so be sure everything works as expected since there's no way back, once overwritten things are gone.

Since you have then a new branch we need to find out how to "connect" it to this PR, or we just open a new PR and close this one.

Sorry for the wall of text, I hope this helps a bit :)

kYc0o commented:
Can you rebase please? This will be reviewed for the next release.

rajma996 commented:
@kYc0o sorry for the delay.

miri64 commented on core/include/tcb.h:
```diff
@@ -0,0 +1,91 @@
+/*
```
This file was removed in master, so please don't re-include it.

miri64 commented:
Postponed due to feature freeze

A-Paul commented:
@rajma996, ping :-)

OlegHahm commented:
superseded by #6407.
